File: golden/ruby/2_sum_of_certain_multiples.rb
def solution(x)
  match = ->(i) { i % 15 != 0 && (i % 3 == 0 || i % 5 == 0) }
  integers = (0...x).to_a.select { |i| match.call(i) }
  return 0 if integers.empty?
  integers.reduce(0) { |a, e| a + e }
end

File: golden/python/2_sum_of_certain_multiples.py
from functools import reduce

def solution(x):
    match = lambda i: (i % 3 == 0 or i % 5 == 0) and i % 15 != 0
    integers = [i for i in range(x) if match(i)]
    if len(integers) == 0:
        return 0

    return reduce((lambda m, i: m + i), integers)
